src/cloudWatchLogs/registry/logDataRegistry.ts

```diff
@@ -364,10 +364,30 @@ export async function getLogEventsFromUriComponents(
         throw new Error('cwl:`getLogEvents` did not return anything.')
     }
 
+    let nextForwardToken = response.nextForwardToken
+    let nextBackwardToken = response.nextBackwardToken
+
+    /**
+     * The getLogEvents() api returns the same token from the request
+     * if there are no more results, but the rest of the code expects
+     * it to be undefined.
+     *
+     * The following conditional sets the token to undefined
+     * if the response token equals the requested.
+     */
+    if (nextToken) {
+        const isForward = nextToken.startsWith('f')
```

Comment:
Seems a little fragile. Pagination tokens should be treated as opaque strings. We might need to pass in another param to know which direction the token is associated with, or possible put this logic somewhere else that does know the direction.

Reply:
We were able to drop this whole function completely, so this code has been removed. I explain why we were able to drop it in another comment. Thanks!